frappe/auth.py

```diff
@@ -454,22 +452,12 @@ def validate_ip_address(user):
 
 	from frappe.core.doctype.user.user import get_restricted_ip_list
 
-	# Only fetch required fields - for perf
-	user_fields = ["restrict_ip", "bypass_restrict_ip_check_if_2fa_enabled"]
-	user_info = (
-		frappe.get_cached_value("User", user, user_fields, as_dict=True)
-		if not frappe.flags.in_test
-		else frappe.db.get_value("User", user, user_fields, as_dict=True)
-	)
-	ip_list = get_restricted_ip_list(user_info)
+	user_info = frappe.get_cached_doc("User", user)
+	ip_list = user_info.get_restricted_ip_list()
 	if not ip_list:
 		return
 
-	system_settings = (
-		frappe.get_cached_doc("System Settings")
-		if not frappe.flags.in_test
-		else frappe.get_single("System Settings")
-	)
+	system_settings = frappe.get_cached_doc("System Settings")
```

Comment:
How does this fare using `frappe.get_system_settings` in the following lines instead? 🐎

Reply:
Ah yes good catch. I just removed in_test branches but didn't look at what it was doing.

Reply:
It would probably fare better because of one less Redis call. System settings aren't fetched by `frappe.cache` after https://github.com/frappe/frappe/pull/28992  :rocket: